src/utils/actions.js

```diff
@@ -230,6 +230,77 @@ const actions = [
     },
     requiredInput: ['partial'],
   },
+  {
+    /** *************************************
+     * Execute Automation contacts internal urls to obtain automation
+     * information and execute the automation selected in the property panel in
+     * edit mode.
+     *
+     * ARGS
+     * app - Reference to current app inherited from index.js
+     * automation - the item id of the automation to contact the items api
+     * automationPostData - boolean value. If true, creates a bookmark of the
+     * current selections and sends the resulting bookmark id as an input
+     * parameter to the selected automation.
+    */
+
+    label: 'Execute Automation',
+    value: 'executeAutomation',
+    getActionCall: ({ app, automation, automationPostData }) => async () => {
+      if (automation !== undefined) {
+        try {
+          const itemInfo = await fetch(`../api/v1/items/${automation}`)
+            .then(response => response.json());
+          const autoInfo = await fetch(`../api/v1/automations/${itemInfo.resourceId}`)
+            .then(response => response.json());
```

Comment:
Is the resourceId different from the automation id? Why not store the resourceId from the start. Otherwise you need to retrieve it every time you execute this automation

Reply:
Yes, the item id is from collections service `${automation}`. The `${itemInfo.resourceId}` is the guid for the automation within automations core service.